Content.Server/_Goobstation/Possession/PossessionSystem.cs

```diff
@@ -0,0 +1,279 @@
+{
+    [Dependency] private readonly IGameTiming _timing = default!;
+    [Dependency] private readonly MobStateSystem _mobState = default!;
+    [Dependency] private readonly SharedPopupSystem _popup = default!;
+    [Dependency] private readonly SharedMindSystem _mind = default!;
+    [Dependency] private readonly SharedAudioSystem _audio = default!;
+    [Dependency] private readonly StunSystem _stun = default!;
+    [Dependency] private readonly SharedTransformSystem _transform = default!;
+    [Dependency] private readonly ContainerSystem _container = default!;
+    [Dependency] private readonly ISharedAdminLogManager _admin = default!;
+    [Dependency] private readonly ActionsSystem _action = default!;
+    [Dependency] private readonly PolymorphSystem _polymorph = default!;
+    [Dependency] private readonly TagSystem _tag = default!;
+
+    private ProtoId<TagPrototype> _cannotSuicideAnyTag = "CannotSuicideAny"; // DeltaV - Don't use literals.
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<PossessedComponent, MapInitEvent>(OnInit);
+        SubscribeLocalEvent<PossessedComponent, ComponentRemove>(OnComponentRemoved);
+
+        SubscribeLocalEvent<PossessedComponent, ExaminedEvent>(OnExamined);
+
+        SubscribeLocalEvent<PossessedComponent, EndPossessionEarlyEvent>(OnEarlyEnd);
+    }
+
+    public override void Update(float frameTime)
+    {
+        base.Update(frameTime);
+
+        var query = EntityQueryEnumerator<PossessedComponent>();
+        while (query.MoveNext(out var uid, out var comp))
+        {
+            if (_timing.CurTime >= comp.PossessionEndTime)
+                RemComp<PossessedComponent>(uid);
+
+            comp.PossessionTimeRemaining = comp.PossessionEndTime - _timing.CurTime;
+        }
+    }
+
+    private void OnInit(Entity<PossessedComponent> possessed, ref MapInitEvent args)
+    {
+        if (!HasComp<WeakToHolyComponent>(possessed))
+            AddComp<WeakToHolyComponent>(possessed).AlwaysTakeHoly = true;
+        else
+            possessed.Comp.WasWeakToHoly = true;
+
+        if (possessed.Comp.HideActions)
+            possessed.Comp.HiddenActions = _action.HideActions(possessed);
+
+        _action.AddAction(possessed, ref possessed.Comp.ActionEntity, possessed.Comp.EndPossessionAction);
+
+        _tag.AddTag(possessed, _cannotSuicideAnyTag); // DeltaV - Use ProtoId.
+
+        possessed.Comp.PossessedContainer = _container.EnsureContainer<Container>(possessed, "PossessedContainer");
+    }
+
+    private void OnEarlyEnd(EntityUid uid, PossessedComponent comp, ref EndPossessionEarlyEvent args)
+    {
+        if (args.Handled)
+            return;
+
+        // if polymorphed, undo
+        _polymorph.Revert(uid);
+        RemCompDeferred(uid, comp);
+
+        args.Handled = true;
+    }
+    private void OnComponentRemoved(Entity<PossessedComponent> possessed, ref ComponentRemove args)
```

Comment:
Blank line between methods
```suggestion
    }

    private void OnComponentRemoved(Entity<PossessedComponent> possessed, ref ComponentRemove args)
```